pkg/lib/encoding/encoding.go

```diff
@@ -0,0 +1,95 @@
+package encoding
+
+import (
+	"bytes"
+	"io"
+
+	"compress/gzip"
+	"encoding/base64"
+)
+
+// EncodeGzipBase64 applies gzip compression to the given bytes,
+// followed by base64 encoding.
+func EncodeGzipBase64(data []byte) ([]byte, error) {
+	gzipped, err := gzipEncode(data)
+	if err != nil {
+		return nil, err
+	}
+
+	base64d, err := base64Encode(gzipped)
+	if err != nil {
+		return nil, err
+	}
+
+	return base64d, nil
+}
+
+// DecodeGzipBase64 applies base64 decoding to the given bytes,
+// followed by gzip decompression.
+func DecodeGzipBase64(data []byte) ([]byte, error) {
+	unbase64d, err := base64Decode(data)
+	if err != nil {
+		return nil, err
+	}
+
+	ungzipped, err := gzipDecode(unbase64d)
+	if err != nil {
+		return nil, err
+	}
+
+	return ungzipped, nil
+}
+
+func gzipEncode(data []byte) ([]byte, error) {
+	buf := &bytes.Buffer{}
+
+	writer := gzip.NewWriter(buf)
+	_, err := writer.Write(data)
```

Comment:
I suppose we don't check the number of bytes written after compression because don't have a way of knowing how big the compressed file should be? I suppose lossy conversion is not a concern in this particular case?

Reply:
> I suppose we don't check the number of bytes written after compression because don't have a way of knowing how big the compressed file should be?

Right, I see nothing useful we could do with this number rather than the spamming the already-noisy logs :-)

> I suppose lossy conversion is not a concern in this particular case?
gzip is loseless - this is an absolute requirement for our compression use here...